packages/core/lib/generators/assets_generator.dart

```diff
@@ -570,7 +570,16 @@ class AssetGenImage {
     );
   }
 
-  ImageProvider provider() => AssetImage(_assetName);
+  ImageProvider provider({
```

Comment:
This is the one and only real change : 
* `provider` method now sets the package if the flag is enabled
* `provider` method now accepts arguments that mirror the `AssetImage` constructor

Note that the rest of the files are only updates of tests and examples